### PR TITLE
TestCafe: Add ServicePage and basic Service test

### PR DESCRIPTION
**Pull request by alexanderturinske**
- Relates to #438

### CODE REVIEW

File: app/components/listing/ActionSidebar.jsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ class ListPageSidebar extends React.Component {
           <li key={action.name}>
             {
               action.to || action.handler
-                ? <Link to={action.to} onClick={action.handler}>
+                ? <Link to={action.to} onClick={action.handler} className={`listting--aside--${action.name.toLowerCase()}`}>
```
Comment (alexanderturinske) [inline app/components/listing/ActionSidebar.jsx]: Two "t"s

File: testcafe/pages/ServicePage.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    const baseSelector = ReactSelector(baseSelectorName);
+    this.serviceName = baseSelector.find('.listing--main--left header');
+    this.description = baseSelector.find('.listing--main--left--about');
+    this.details = baseSelector.find('.listing--main--left--details');
+    // TODO: Can't use nested React component name PhoneNumber because it is
+    // instantiated in both the header and the body of the page and because the
+    // testcafe-react-selectors plugin is currently unable to mix CSS selectors
+    // in between React component names.
+    // https://github.com/DevExpress/testcafe-react-selectors/issues/51
+    this.phones = baseSelector.find('..listing--main--left--info');
```
Comment (alexanderturinske) [inline testcafe/pages/ServicePage.js]: Two "."

File: testcafe/pages/ServicePage.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    this.editButton = baseSelector.find('.listing--aside-edit');
```
Comment (alexanderturinske) [inline testcafe/pages/ServicePage.js]: Single "-"

File: testcafe/service.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import ServicePage from './pages/ServicePage';
+
+const servicePage = new ServicePage();
+
+fixture `Service Page`
+  .page(servicePage.url(1));
+
+test('Confirm Page Loads with Information', async (t) => {
+  const serviceName = 'Women\'s clinicA service offered by New Resource Name';
```
Comment (alexanderturinske) [inline testcafe/service.js]: This works on local, but not the testing suite. Going to switch to check for existence.

File: testcafe/pages/ServicePage.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    const baseSelectorName = 'ServicePage';
+    const baseSelector = ReactSelector(baseSelectorName);
+    this.serviceName = baseSelector.find('.listing--main--left header');
+    this.description = baseSelector.find('.listing--main--left--about');
+    this.details = baseSelector.find('.listing--main--left--details');
+    // TODO: Can't use nested React component name PhoneNumber because it is
+    // instantiated in both the header and the body of the page and because the
+    // testcafe-react-selectors plugin is currently unable to mix CSS selectors
+    // in between React component names.
+    // https://github.com/DevExpress/testcafe-react-selectors/issues/51
```
Comment (richardxia) [inline testcafe/pages/ServicePage.js]: That ticket is out of date. You can now nest both CSS and React selectors under other ReactSelectors with the `.findReact()` method: https://github.com/DevExpress/testcafe-react-selectors#searching-for-nested-components

This also means that you don't need to add all the CSS classes that you added just for the sake of selecting components for tests.

Comment (alexanderturinske) [reply testcafe/pages/ServicePage.js]: The problem I have with that is then the tests are dependent on the layout of the HTML; by relying on classes, those elements can move anywhere on the page and the tests won't break. For example, the line below (editButton) would turn into `this.editButton = baseSelector.findReact('ListPageSidebar').findReact('ul').nth(0)` if I didn't have the class.

Comment (richardxia) [reply testcafe/pages/ServicePage.js]: Okay, yeah, that's a fair comment, though I feel like our CSS names are a bit too generic and there's a minor risk of accidentally selecting something else. But we can deal with that later.

**Pull request merged**